// [[Reason: Semantically invalid. TSC does not emit .d.ts]] ////

//// [tests/cases/compiler/genericDefaultsErrors.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,5 +1,47 @@
 
+
+//// [genericDefaultsErrors.d.ts]
+declare const x: any;
+declare function f03<T extends string = number>(): void;
+declare function f04<T extends string, U extends number = T>(): void;
+declare function f05<T, U extends number = T>(): void;
+declare function f06<T, U extends T = number>(): void;
+declare function f11<T, U, V = number>(): void;
+declare function f12<T, U = T>(a?: U): void;
+interface i00<T> {
+}
+interface i00<U = number> {
+}
+interface i01<T = number> {
+}
+interface i01<T = string> {
+}
+interface i04<T = number, U> {
+}
+interface i05<T extends string = number> {
+}
+interface i06<T extends string, U extends number = T> {
+}
+interface i07<T, U extends number = T> {
+}
+interface i08<T, U extends T = number> {
+}
+interface i09<T, U, V = number> {
+}
+type i09t00 = i09;
+type i09t01 = i09<1>;
+type i09t02 = i09<1, 2>;
+type i09t03 = i09<1, 2, 3>;
+type i09t04 = i09<1, 2, 3, 4>;
+interface i10 {
+    x: T;
+}
+interface i10<T = number> {
+}
+interface SelfReference<T = SelfReference> {
+}
+//# sourceMappingURL=genericDefaultsErrors.d.ts.map
 /// [Errors] ////
 
 genericDefaultsErrors.ts(3,41): error TS2344: Type 'number' does not satisfy the constraint 'string'.
 genericDefaultsErrors.ts(4,59): error TS2344: Type 'T' does not satisfy the constraint 'number'.
